### PR TITLE
nsmgr: correctly pre-create pinns directories

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

> /kind api-change
/kind bug
> /kind ci
> /kind cleanup
> /kind dependency-change
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake
> /kind other

#### What this PR does / why we need it:
https://github.com/cri-o/cri-o/pull/4532 had a silly bug where we removed and recreated the incorrect directory :upside_down_face: 

while here, I also moved the responsibility of creating the correct directories to nsmgr, as i think it makes more sense
#### Does this PR introduce a user-facing change?

```release-note
none
```

### CODE REVIEW

File: internal/config/nsmgr/nsmgr.go [hunk 1]
```diff
@@ -143,3 +143,9 @@ func getSysctlForPinns(sysctls map[string]string) string {
 	}
 	return strings.TrimSuffix(g.String(), pinnsSysctlDelim)
 }
+
+// NamespacedirForType returns the sub-directory for that particular NSType
+// which is of the form `$namespaceDir/$nsType+"ns"`
+func (mgr *NamespaceManager) NamespaceDirForType(ns NSType) string {
```
Comment (kolyshkin) [inline internal/config/nsmgr/nsmgr.go]: Can probably be just called `Dir`.

Comment (kolyshkin) [reply internal/config/nsmgr/nsmgr.go]: or `Path`

Comment (haircommander) [reply internal/config/nsmgr/nsmgr.go]: went for dirForType

**Comment by haircommander** [comment]
@saschagrunert @giuseppe @mrunalp @umohnani8 @QiWang19 @wgahnagl PTAL

**Review by saschagrunert — APPROVED** [review]
LGTM

**Review by mrunalp — APPROVED** [review]
/lgtm

**Pull request merged**